core_v2/sources/v2_domains.move

```diff
@@ -0,0 +1,1225 @@
+
+        assert!(
+            is_name_registerable(domain_name, option::some(subdomain_name)),
+            error::invalid_state(ENAME_NOT_AVAILABLE)
+        );
+
+        // We are registering a subdomain name: this has no cost, but is only doable by the owner of the domain
+        validate_name_string(subdomain_name);
+
+        // Ensure signer owns the domain we're registering a subdomain for
+        let signer_addr = signer::address_of(sign);
+        assert!(
+            is_owner_of_name(signer_addr, option::none(), domain_name),
+            error::permission_denied(ENOT_OWNER_OF_DOMAIN)
+        );
+
+        let registration_duration_secs = expiration_time_sec - timestamp::now_seconds();
+
+        let price = v2_price_model::price_for_subdomain(registration_duration_secs);
+        coin::transfer<AptosCoin>(sign, v2_config::fund_destination_address(), price);
+
+        register_name_internal(sign, option::some(subdomain_name), domain_name, registration_duration_secs, price);
+    }
+
+    /// Router-only registration that does not take registration fees. Should only be used for v1=>v2 migrations.
+    /// We skip checking registration duration because it is not necessarily a whole number year
+    public fun register_name_with_router(
+        router_signer: &signer,
+        sign: &signer,
+        domain_name: String,
+        subdomain_name: Option<String>,
+        registration_duration_secs: u64,
+    ) acquires CollectionCapability, NameRecord, RegisterNameEvents, ReverseRecord, SetReverseLookupEvents {
+        assert!(address_of(router_signer) == @router_signer, error::permission_denied(ENOT_ROUTER));
+        // For subdomains, this will check that the domain exists first
+        assert!(is_name_registerable(domain_name, subdomain_name), error::invalid_state(ENAME_NOT_AVAILABLE));
+        if (option::is_some(&subdomain_name)) {
+            validate_name_string(*option::borrow(&subdomain_name));
+        } else {
+            validate_name_string(domain_name);
+        };
+        register_name_internal(sign, subdomain_name, domain_name, registration_duration_secs, 0);
+        // No automatic assignment of primary name / target_addr. These are handled by the router
+    }
+
+    /// Register a name. Accepts an optional subdomain name, a required domain name, and a registration duration in seconds.
+    /// For domains, the registration duration is only allowed to be in increments of 1 year, for now
+    /// Since the owner of the domain is the only one that can create the subdomain, we allow them to decide how long they want the underlying registration to be
+    /// The maximum subdomain registration duration is limited to the duration of its parent domain registration
+    ///
+    /// NOTE: Registration validation already done. This function does not perform any validation on whether `sign` is allowed to register this name
+    fun register_name_internal(
+        sign: &signer,
```

Comment:
can we label this instead to buyer or purchaser or creator? `sign` is not a very descriptive name for the actor here.